### DISCUSSION TITLE
Authorizer Discussions!

### DISCUSSION THREAD

**Discussion by tejado** [tejado/Authorizer#49] [Announcements]
## 👋 Welcome!
  We’re using Discussions as a place to connect with other members of our community. We hope that you:
  * Ask questions you’re wondering about.
  * Share ideas.
  * Engage with other community members.
  * Welcome others and are open-minded. Remember that this is a community we
  build together 💪.

  To get started, comment below with an introduction of yourself and tell us about what you do with this community.

**Comment by jmichael2497** [comment]
hi, i'm a casual tech and open source enthusiast, with background in tech support and basic programming fundamentals (that are rarely used beyond trying to provide useful feedback in bug reports), and just wanted to share some feedback and compliments:

using the combination of https://github.com/tejado/android-usb-gadget to add function or gadget (one or both may work depending on device used) for HID keyboard and mouse, but still need an app to make use of those.

* LG v20 H918 with LOS A12 build 20230713 kernel 4.4.302
[n] add function
[n] add gadget

* Moto x4 XT1900-1 with LOS A13 build 20230711 kernel 4.4.302
[y] add function
[y] add gadget

using https://github.com/tejado/Authorizer (pwsafe plus autotype) with my only device that made it past setup task (Moto x4) successfully used to autotype over USB *and* bluetooth to W10-64 22h2 dell laptop from 2013! (only works switched to add new gadget, not with add function to pre-existing default multi-gadget)

(NOTE: to workaround stupid MS bug of undersized frame that cannot be resized and cuts off bluetooth pairing numbers for confirmation... had to start pairing from W10 side, instead of app side.)

so if that app can autotype over USB and BT, then maybe a skilled developer ;^D can be encouraged to make a replacement typing window (to use with whatever keyboard, such as from other pw managers or full https://github.com/klausw/hackerskeyboard) and mouse app?

the usually mentioned (but abandoned) old https://github.com/pelya/android-keyboard-gadget/tree/master/compiled/USB-Keyboard-app hangs at starting hidg0 and hidg1 screen on my only testable device (Moto x4).
